src/asm/section.cpp

```diff
@@ -471,6 +471,27 @@ uint32_t sect_GetOutputOffset(void)
 	return curOffset + loadOffset;
 }
 
+// Returns how many bytes need outputting for the specified alignment and offset to succeed
+uint32_t sect_GetAlignBytes(uint8_t alignment, uint16_t offset)
+{
+	struct Section *sect = sect_GetSymbolSection();
+
+	// If the section is undefined, or not fixed and not aligned, no bytes are needed
+	if (!sect || (sect->org == (uint32_t)-1 && sect->align == 0))
+		return 0;
+
+	uint32_t pcValue;
+
+	if (sect->org != (uint32_t)-1)
+		pcValue = sym_GetPCValue();
+	else if (sect->align != 0)
+		pcValue = (sect->alignOfs + curOffset) % (1 << sect->align);
+	else
+		pcValue = 0;
+	// We need `(pcValue + return value) % (1 << alignment) == offset
+	return (offset - pcValue) % (1 << alignment);
```

Comment:
Can't that modulo's LHS become negative? I don't think we'd handle that well. We can probably avoid that with this:

```suggestion
	return static_cast<uint16_t>(offset - pcValue) % (1 << alignment);
```

cc @aaaaaa123456789 to double-check that the semantics look correct, and also fuck integer promotion.

Reply:
Integer promotion won't screw you over because you're converting to an unsigned type. If `uint32_t` is at least as large as `int`, the value in parens is unsigned already. If it is not, the value in parens is `int`, but converting from `int` to `uint16_t` is guaranteed to do a proper-overflow two's-complement conversion.

Reply:
Yeah, but it wouldn't have without the cast, wherein the modulo would have been applied to `int`. (Also, both `offset` and `pcValue` are `uint16_t`s, aren't they?)

But then you confirm the semantics *with the cast* are correct, yes?

Reply:
`pcValue` is `uint32_t` (declared a bit above). But yes, with the cast, the semantics are correct, even if the final result will actually be `int`.